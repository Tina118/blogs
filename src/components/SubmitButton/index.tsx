import { FC } from 'react'

interface SubmitButtonProps {
  buttonText: string
  onClick: () => void
  [rest:keyof any]: any
}

const SubmitButton: FC<SubmitButtonProps> = ({
  buttonText,
  onClick,
  ...rest
}) => (
  <div className="text-center text-lg-start mt-4 pt-2">
    <button
      type="button"
      className="btn btn-primary btn-lg"
      onClick={onClick}
      {...rest}
    >
      {buttonText}
    </button>
  </div>
)

export default SubmitButton
