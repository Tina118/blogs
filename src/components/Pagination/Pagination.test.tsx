import { render } from '@testing-library/react';
import SubmitButton from './index';

test('renders main App', () => {
  const fragment = render(<SubmitButton numberOfTotalPages={10} handlePagination={()=>{}} />);
  expect(fragment).toMatchSnapshot();
});
