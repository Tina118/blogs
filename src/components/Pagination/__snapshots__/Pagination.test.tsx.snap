// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center align-items-center"
      >
        <ul
          class="pagination"
        >
          <li
            class="page-item disabled"
          >
            <a
              class="page-link"
              href="#"
            >
              Previous
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link active"
              href="#"
            >
              1
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              2
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              3
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              4
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              5
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              6
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              7
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              8
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              9
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link "
              href="#"
            >
              10
            </a>
          </li>
          <li
            class="page-item "
          >
            <a
              class="page-link"
              href="#"
            >
              Next
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center align-items-center"
    >
      <ul
        class="pagination"
      >
        <li
          class="page-item disabled"
        >
          <a
            class="page-link"
            href="#"
          >
            Previous
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link active"
            href="#"
          >
            1
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            2
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            3
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            4
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            5
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            6
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            7
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            8
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            9
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link "
            href="#"
          >
            10
          </a>
        </li>
        <li
          class="page-item "
        >
          <a
            class="page-link"
            href="#"
          >
            Next
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
